### PR TITLE
Filemanager external api e2e

### PR DESCRIPTION
**Pull request by ioedeveloper**
Fix https://github.com/ethereum/remix-ide/issues/2911

### CODE REVIEW

File: src/app/files/fileManager.js [hunk 1]
```diff
@@ -194,10 +194,24 @@ class FileManager extends Plugin {
...
    */
   async rename (oldPath, newPath) {
-    await this.__handleExists(oldPath, `Cannot rename ${oldPath}`)
+    await this._handleExists(oldPath, `Cannot rename ${oldPath}`)
     const isFile = await this.isFile(oldPath)
+    const newPathExists = await this.exists(newPath)
+    const provider = this.fileProviderOf(oldPath)
 
-    this.fileRenamedEvent(oldPath, newPath, !isFile)
+    if (isFile) {
```
Comment (yann300) [inline src/app/files/fileManager.js:202]: looks like `fileRenameEvent` is not called and that the event located at https://github.com/ethereum/remix-ide/pull/2916/files#diff-f28e4232b4dbf7d53c7b5543bdc9ad4fR314 will never be trigerred

Comment (ioedeveloper) [reply src/app/files/fileManager.js:202]: `fileRenamedEvent` is triggered after rename from provider. https://github.com/ethereum/remix-ide/blob/1b2bac5f1b84254731e17bebecc90ab2076e6f8a/src/app/files/fileManager.js#L270
So the changes are in order.

Comment (ioedeveloper) [reply src/app/files/fileManager.js:202]: Adding a call to the `fileRenamedEvent` event in rename function triggers it twice. You can try it out locally.

Comment (yann300) [reply src/app/files/fileManager.js:202]: does that mean it was actually a bug?

Comment (ioedeveloper) [reply src/app/files/fileManager.js:202]: Yes, it was.

**Pull request merged**